cluster-autoscaler/processors/customresources/dra_processor_test.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package customresources
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	resourceapi "k8s.io/api/resource/v1beta1"
+	"k8s.io/autoscaler/cluster-autoscaler/simulator/clustersnapshot/store"
+	"k8s.io/autoscaler/cluster-autoscaler/simulator/clustersnapshot/testsnapshot"
+	drasnapshot "k8s.io/autoscaler/cluster-autoscaler/simulator/dynamicresources/snapshot"
+	"k8s.io/autoscaler/cluster-autoscaler/simulator/framework"
+
+	"github.com/stretchr/testify/assert"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	testprovider "k8s.io/autoscaler/cluster-autoscaler/cloudprovider/test"
+	"k8s.io/autoscaler/cluster-autoscaler/context"
+)
+
+var (
+	node1Slice1 = createNodeResourceSlice("node1", "local-slice-1", "driver_X", "pool", []string{"device_A", "device_B"})
+	node1Slice2 = createNodeResourceSlice("node1", "local-slice-1", "driver_Y", "pool", []string{"device_A", "device_C"})
+	node2Slice1 = createNodeResourceSlice("node2", "local-slice-2", "driver_Y", "pool", []string{"device_A", "device_C"})
+	node2Slice2 = createNodeResourceSlice("node2", "local-slice-2", "driver_X", "pool", []string{"device_A", "device_B"})
+	node3Slice1 = createNodeResourceSlice("node3", "local-slice-3", "driver_X", "pool", []string{"device_A", "device_B"})
+	node3Slice2 = createNodeResourceSlice("node3", "local-slice-3", "driver_Y", "pool", []string{"device_A", "device_B"})
+)
+
+func createNodeResourceSlice(nodeName, id, driverName, poolName string, devicesNames []string) *resourceapi.ResourceSlice {
+	devices := []resourceapi.Device{}
+	for _, name := range devicesNames {
+		devices = append(devices, resourceapi.Device{Name: name})
+	}
+	spec := resourceapi.ResourceSliceSpec{NodeName: nodeName, Driver: driverName, Pool: resourceapi.ResourcePool{Name: poolName}, Devices: devices}
+	return &resourceapi.ResourceSlice{ObjectMeta: metav1.ObjectMeta{Name: id}, Spec: spec}
+}
+func TestFilterOutNodesWithUnreadyDRAResources(t *testing.T) {
```

Comment:
Could you convert this test to be table-based so that we can test different kinds of Nodes (e.g. non-DRA Node, DRA Node with a single pool (ready/not ready), DRA Node with multiple pools (ready/all pools not ready/only some pools ready), multiple different kinds of Nodes)?